### PR TITLE
[#203] add global config for irb flags

### PR DESCRIPTION
**Pull request by tecden**

### CODE REVIEW

**Comment by tecden** [comment]
I tested with a Rails 7 application that the --nomultiline flag from the irb config is also correctly passed on to the console in the staging instance :heavy_check_mark:

File: lib/geordi/util.rb [hunk 1]
```diff
@@ -89,14 +89,19 @@ def console_command(environment)
...
-          "#{binstub_or_fallback('rails')} console -e #{environment} #{use_multiline_flag}"
+          "#{binstub_or_fallback('rails')} console -e #{environment} #{irb_flags}"
         end
       end
 
+      def irb_flags
+        irb_flags = [use_multiline_flag, Settings.new.irb_flags].join(' ').strip
+        "-- #{irb_flags}" unless irb_flags.empty?
+      end
+
```
[the remark this answers is not in the document]
Comment (tecden) [reply lib/geordi/util.rb:100]: Personally, I think that the code is much more readable if the logic for the IRB flags sits in its own method. 
What do you think about putting the methods `#irb_flag` and `#use_multiline_flag` in the private scope of the Util class?

File: lib/geordi/commands/console.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 Selecting the server: `geordi console staging -s` shows a menu with all available
 servers. When passed a number, directly connects to the selected server.
+Irb flags defined in the global Geordi config file (`~/.config/geordi/global.yml`),
+are automatically passed to the irb console.
```
[the remark this answers is not in the document]
Comment (tecden) [reply lib/geordi/commands/console.rb:10]: do you mean Geordi::Settings::GLOBAL_SETTINGS_FILE_NAME ?
I don't think it's a good idea to write this in the long description. The constant is prefixed with ENV['HOME']. And then my path is written into the global readme?

Or did you mean a different constant?

File: lib/geordi/util.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (codener) [reply lib/geordi/util.rb:105]: As discussed, please implement as suggested.

File: lib/geordi/commands/console.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (codener) [reply lib/geordi/commands/console.rb:11]: Oh, that's a good point. In this case, just leave it like this.

File: lib/geordi/util.rb [hunk 3]
```diff
@@ -94,14 +94,15 @@ def console_command(environment)
...
         else
-          "#{binstub_or_fallback('rails')} console -e #{environment} #{use_multiline_flag}"
-        end
-      end
+          use_multiline = if irb_version >= Gem::Version.new('1.2') && ruby_version < Gem::Version.new('3.0')
+            Interaction.note 'Using --nomultiline switch for faster pasting'
+            '--nomultiline'
+          end
+
+          irb_flags = [use_multiline, Settings.new.irb_flags].join('').strip
```
Comment (codener) [inline lib/geordi/util.rb:102]: Wouldn't you need to join with a `' '` space?! There should be a test covering this.

File: features/console.feature [hunk 4]
```diff
@@ -1,6 +1,21 @@
...
+      And a file named "config/deploy.rb" with "deploy file exists"
+      And a file named "config/deploy/staging.rb" with:
+      """
+      set :deploy_to, '/var/www/example.com'
+      set :user, 'user'
+      server 'www.example.com'
+      """
+    When I run `geordi console staging`
+    Then the output should contain "# Opening a Rails console on staging"
+      And the output should contain "Util.run! ssh, user@www.example.com, -t, cd /var/www/example.com/current && bash --login -c 'bundle exec rails console -e  -- --foo --baz"
```
Comment (codener) [inline features/console.feature:16]: There is the environment missing after `-e` (shouldn't it be "staging"?), and/or a double space present. Please show how `--nomultiline` is integrated.

File: lib/geordi/commands/console.rb [hunk 5]
```diff
@@ -6,6 +6,9 @@
 
 Selecting the server: `geordi console staging -s` shows a menu with all available
 servers. When passed a number, directly connects to the selected server.
+
+IRB flags defined in the global Geordi config file (`~/.config/geordi/global.yml`),
+are automatically passed to the irb console.
```
Comment (codener) [inline lib/geordi/commands/console.rb:11]: We must show how to use the setting:
```suggestion
IRB flags can be given as `irb_flags: '...'` in the global Geordi config file
(`~/.config/geordi/global.yml`).
```

File: features/console.feature (hunk 4 above)
Comment (tecden) [reply features/console.feature:16]: You are right. The environment is pulled out of the capistrano config. In the test setup the `rails_env` is missing in `config/deploy/staing.rb`. I have adjusted the test setup accordingly.

File: lib/geordi/commands/console.rb [hunk 6]
```diff
@@ -7,8 +7,7 @@
 Selecting the server: `geordi console staging -s` shows a menu with all available
 servers. When passed a number, directly connects to the selected server.
 
-IRB flags defined in the global Geordi config file (`~/.config/geordi/global.yml`),
-are automatically passed to the irb console.
+IRB flags can be given as `irb_flags: '...' in the global Geordi config file (`~/.config/geordi/global.yml`).
```
Comment (codener) [inline lib/geordi/commands/console.rb:10]:
```suggestion
IRB flags can be given as `irb_flags: '...'` in the global Geordi config file (`~/.config/geordi/global.yml`).
```
Tip: review README changes (needs update!) in the Markdown preview.

**Review by codener — APPROVED** [review]

**Pull request merged**